atom/common/deprecate_util.cc

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2019 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "atom/common/deprecate_util.h"
+
+namespace atom {
+
+// msg/type/code arguments match Node's process.emitWarning() method. See
+// https://nodejs.org/api/process.html#process_process_emitwarning_warning_type_code_ctor
+v8::Maybe<bool> EmitDeprecationWarning(node::Environment* env,
+                                       std::string warning_msg,
```

Comment:
Should pass `const std::string&`.

It seems that this API is mostly called with string literals, if it is the case then passing `base::StringPiece` would be most efficient.